akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala

```diff
@@ -124,11 +159,13 @@ object EventEnvelope {
  * API May Change
  */
 @ApiMayChange
-final class EventEnvelope[Event](
+final class EventEnvelope[Event] private (
     val offset: Offset,
     val persistenceId: String,
     val sequenceNr: Long,
-    val eventOption: Option[Event],
+    private val _eventOption: Option[Event],
+    /** INTERNAL API */
+    @InternalStableApi private[akka] val serializedEvent: Option[EventEnvelope.SerializedEvent],
```

Comment:
I have updated to more explicit `SerializedEvent`. We should be able keep this as an internal feature and deserialize before handing out to user code.